### ISSUE TITLE
render() causes OptiX API error = 7050 (Launch failure)

### ISSUE THREAD

**Issue by Samia067** [mitsuba-renderer/mitsuba2#210]
Hi. I am trying to render a scene with gpu_autodiff_rgb. However, when performing the render() command, I get this error:
rt_check(): OptiX API error = 7050 (Launch failure) in ../src/librender/scene_optix.inl:521.

There were no errors in compilation. I am using Ubuntu 18.04, cuda 10.2, clang-9, clang++-9 and optix 7.1. To check if it was a problem with Optix installation, I ran sample programs provided with the Optix SDK and they compiled and ran properly. 

Any ideas what is going wrong here?
Thank you for the help!

**Comment by Speierers** [comment]
Hi @Samia067 ,

Does this happen when calling `render()` from a Python script or with the `mistuba` command in the terminal?

**Comment by Samia067** [comment]
It happens with a python script.

**Comment by Speierers** [comment]
This would be a good test to assert whether the problem comes from your setup or the script. Could you give it a try?

**Comment by Samia067** [comment]
I tested running mitsuba from the terminal. The following is the command I used to render a provided example. 
  ./build/dist/mitsuba -m gpu_autodiff_rgb ./resources/data/scenes/cbox/cbox-rgb.xml

The good news is it does not give the same Optix error as the console. However, I do face a cuda memory allocation error. This is the error message:
cuda_check(): runtime API error = 0002 "cudaErrorMemoryAllocation" in ../ext/enoki/src/cuda/horiz.cu:59.

**Comment by Speierers** [comment]
Could you please share with us the specs of your GPU?

Also what is the resolution of the render your computing in python? And the number of samples per pixel?

**Comment by Samia067** [comment]
I am using a GeForce RTX 2080 Ti. It has a memory capacity of 11GB. 

In python, the resolution of the render is 765x405 and the spp is 25. This low spp is for testing purposes only. I plan to use a higher spp later for better image outputs.

**Comment by Speierers** [comment]
Do you still get the error when using only 1 spp in python as well?

We are aware that the current version of Mitsuba 2 uses a lot of memory on the GPU and are working hard to reduce this in a future release.

**Comment by Samia067** [comment]
It works from python with a sample count of 1. I tried increasing it to 16 and it still works but increasing it beyond that does not work. It gives the Optix launch error. 

From the terminal using the mitsuba command, it works upto a sample per pixel size of 128 and gives a cuda memory allocation error beyond that. 

How can I render images with higher quality than 16 spp from the python console?

**Comment by Speierers** [comment]
You can use the `samples_per_pass` property of the integrators to split the rendering in multiple passes. What this does under the hood is simply rendering the scene with a lower SPP count multiple times (using different seed) and averaging the resulting images.

For optimization / gradient descent, you can do the same thing: e.g. take more iterations with a lower SPP count and a lower learning rate.

It is also possible to render the "forward" image with multiple passes in order to get a "cleaner" image for computing the loss while maintaining a lower SPP count for the backward pass. Take a look at the doc for the `autodiff.py:render()` function, especially the ` unbiased` and `spp` argument.

**Comment by Samia067** [comment]
Thanks!
